name.abuchen.portfolio/src/name/abuchen/portfolio/model/PortfolioTransaction.java

```diff
@@ -211,7 +214,15 @@ public Quote getGrossPricePerShare(CurrencyConverter converter)
         // transaction currency and not in security currency) we must convert
         // the gross value (instead of checking the unit type GROSS_VALUE)
 
-        long grossValue = getGrossValue().with(converter.at(getDateTime())).getAmount();
+        // use exchange rate used within the transaction,
+        // not the historical exchange rate
+        Optional<Unit> grossValueUnit = getUnit(Unit.Type.GROSS_VALUE);
+        BigDecimal exchangeRate = grossValueUnit.isPresent() ? grossValueUnit.get().getExchangeRate()
```

Comment:
Ich füge noch ein ```&& grossValueUnit.get().getForex().getCurrencyCode().equals(converter.getTermCurrency())``` hinzu denn ich denke nicht dass man annehmen kann, das die Zielwährung des Converters immer zwingend die Fremdwährung der Buchung sein muss.

Reply:
Und: wenn ich mir es so recht überlege, sollte doch die ```grossValueUnit``` die korrekten Wert in Zielwährung schon haben, oder?